src/meta/expr.lean

```diff
@@ -618,6 +618,129 @@ meta def is_eta_expansion (val : expr) : tactic (option expr) := do
 
 end expr
 
+/-! ### Declarations about `expr` manipulations -/
```

Comment:
I'd consider putting this in its own file, with an expanded module doc that shows some examples of using `expr_lens`.

Reply:
Sounds good to me.